.github/workflows/c-cpp.yml

```diff
@@ -52,21 +62,49 @@ jobs:
         export -p
         echo '${{ toJSON(matrix) }}'
 
-    - name: Install dependencies
+    - if: ${{ matrix.build.name == 'macos-meson' }}
       run: |
-        sudo apt-get install -y \
-          libphysfs-dev \
+        pip3 install meson ninja
+
+    - if: ${{ contains(matrix.build.os, 'macos') }}
+      name: Install dependencies
+      run: |
+        brew install \
+          pkg-config \
+          sdl2_ttf \
+          sdl2_mixer \
+          ${{ matrix.build.install_packages }}
+
+    - if: ${{ contains(matrix.build.os, 'ubuntu') }}
+      name: Install dependencies
+      run: |
+        sudo apt install -y \
           libsdl2-dev \
           libsdl2-mixer-dev \
           libsdl2-ttf-dev \
-          meson \
-          ninja-build \
           python3 \
-          python3-setuptools \
-          scons
+          python3-setuptools
+
+    - if: ${{ contains(matrix.build.os, 'ubuntu') && contains(matrix.build.name, 'meson') }}
+      name: Install dependencies
+      run: |
+        sudo apt install -y \
+          libphysfs-dev \
+          meson \
+          ninja-build
         # sudo -H python3 -m pip install meson ninja
 
-    - name: Build with Meson
+    - if: ${{ contains(matrix.build.os, 'ubuntu') && contains(matrix.build.name, 'scons') }}
+      name: Install dependencies
+      run: |
+        sudo apt install -y scons
+
+    - if: ${{ contains(matrix.build.name, 'meson') }}
+      name: Meson build
       run: |
         meson setup _build
         ninja -v -C _build
+
+    - if: ${{ contains(matrix.build.name, 'scons') }}
+      name: Scons build
+      run: scons -j 3
```

Comment:
was 3 shown to be the optimal value here or can it be filled in from a variable?

Reply:
No `nproc` on macos. One way to do it is https://stackoverflow.com/a/45181694 but I figured it should be fine to specify 3 there. https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources